python/grass/jupyter/region.py

```diff
@@ -110,27 +111,44 @@ def _set_bbox(self, env):
 class RegionManagerFor2D:
     """Region manager for 2D displays (gets region from display commands)"""
 
-    def __init__(self, use_region, saved_region, env):
+    def __init__(self, use_region, saved_region, width, height, env):
         """Manages region during rendering.
 
         :param use_region: if True, use either current or provided saved region,
                           else derive region from rendered layers
         :param saved_region: if name of saved_region is provided,
                             this region is then used for rendering
+        :param width: rendering width
+        :param height: rendering height
         :param env: environment for rendering
         """
         self._env = env
+        self._width = width
+        self._height = height
         self._use_region = use_region
         self._saved_region = saved_region
         self._extent_set = False
         self._resolution_set = False
+        self._size_set = False
 
     def set_region_from_env(self, env):
         """Copies GRASS_REGION from provided environment
         to local environment to set the computational region"""
         if "GRASS_REGION" in env:
             self._env["GRASS_REGION"] = env["GRASS_REGION"]
 
+    def adjust_rendering_size_from_region(self):
+        """Sets the environmental render width and height variables
+        based on the region dimensions. Only first call of this
+        method sets the variables, subsequent calls do not adjust them.
+        """
+        if not self._size_set:
```

Comment:
Add `self._extent_set` to this function. If the extent is not set yet, this should be recomputed. I think that `if not self._extent_set`, self._set_size should not be set to True.

No error in the current code since `self._extent_set` is True before this is called, but I think it should work as expected even if the functions are called out of order.